driver/src/logging.rs

```diff
@@ -21,47 +27,64 @@ pub fn init(filter: impl AsRef<str>) -> (Logger, GlobalLoggerGuard) {
     (logger, guard)
 }
 
-pub struct CustomFormatter<D: Decorator> {
-    decorator: D,
+/// Uses one decorator for `Error` and `Critical` log messages and the other for
+/// the rest.
+pub struct CustomFormatter<T0, T1> {
+    err_decorator: T0,
+    rest_decorator: T1,
 }
 
-impl<D: Decorator> CustomFormatter<D> {
-    fn new(decorator: D) -> Self {
-        Self { decorator }
+impl<T0, T1> CustomFormatter<T0, T1> {
+    fn new(err_decorator: T0, rest_decorator: T1) -> Self {
+        Self {
+            err_decorator,
+            rest_decorator,
+        }
     }
 }
 
-impl<D: Decorator> Drain for CustomFormatter<D> {
+impl<T0: Decorator, T1: Decorator> Drain for CustomFormatter<T0, T1> {
     type Ok = ();
     type Err = std::io::Error;
     fn log(
         &self,
         record: &Record,
         values: &OwnedKVList,
     ) -> std::result::Result<Self::Ok, Self::Err> {
-        self.decorator.with_record(record, values, |mut decorator| {
-            decorator.start_timestamp()?;
-            slog_term::timestamp_utc(&mut decorator)?;
+        match record.level() {
+            Level::Error | Level::Critical => log_to_decorator(&self.err_decorator, record, values),
+            _ => log_to_decorator(&self.rest_decorator, record, values),
+        }
+    }
+}
 
-            decorator.start_whitespace()?;
-            write!(decorator, " ")?;
+fn log_to_decorator(
+    decorator: &impl Decorator,
+    record: &Record,
+    values: &OwnedKVList,
+) -> std::result::Result<(), std::io::Error> {
+    decorator.with_record(record, values, |mut decorator| {
+        decorator.start_timestamp()?;
+        slog_term::timestamp_utc(&mut decorator)?;
 
-            decorator.start_level()?;
-            write!(decorator, "{}", record.level())?;
+        decorator.start_whitespace()?;
+        write!(decorator, " ")?;
 
-            decorator.start_whitespace()?;
-            write!(decorator, " ")?;
+        decorator.start_level()?;
+        write!(decorator, "{}", record.level())?;
 
-            write!(decorator, "[{}]", record.module())?;
+        decorator.start_whitespace()?;
+        write!(decorator, " ")?;
 
-            decorator.start_whitespace()?;
-            write!(decorator, " ")?;
+        write!(decorator, "[{}]", record.module())?;
 
-            decorator.start_msg()?;
-            writeln!(decorator, "{}", record.msg())?;
-            decorator.flush()?;
+        decorator.start_whitespace()?;
+        write!(decorator, " ")?;
 
-            Ok(())
-        })
-    }
+        decorator.start_msg()?;
+        writeln!(decorator, "{}", record.msg())?;
+        decorator.flush()?;
```

Comment:
It flushes on every message :tada: So no worries about mangling the streams I think.